### ISSUE TITLE
File expiration

### ISSUE THREAD

**Issue by TyroneSama** [WeebDev/loli-safe#30]
It would be nice to be able to configure automatic file deletion, to use lolisafe as a temporary uploader or save on disk space.

The usual way of doing it is "delete after X days", but it could also be useful to have files delete themselves after X days of not being accessed, so that frequently accessed files aren't suddenly pulled down.

**Comment by Pitu** [comment]
I'm not sure I'm fond of the idea. I can see it being useful in the scenario you depicted, but it would take a revamp of the frontend prompting the user if the files should expire or not. I'll think about it and let you know in a few days.

Thanks for the suggestion!

**Comment by alucard0134** [comment]
it would benefit safe.moe extremely

**Comment by Pitu** [comment]
who dat

**Comment by TyroneSama** [comment]
I wasn't really thinking of a per-file option, actually, though that could be useful. I'd personally get more use out of a global setting that applies to all uploads, which at least saves you the trouble of the UI redesign.

**Comment by RyoshiKayo** [comment]
You can just set this up as a cron on the server itself:

```cron
@daily /usr/bin/find /full/path/to/lolisafe/uploads -type f -atime +30 -delete
```
This will delete all files that have not been accessed in over 30 days, not including those that are exactly 30 days.

**Comment by ev1l0rd** [comment]
@RyoshiKayo That wouldn't clean the database references to those files.

While it would solve the immediate issue of storage space, it is a hamfisted method.

**Comment by alucard0134** [comment]
yeah you'd need a check script rather then trying to tie it all into one cronjob, sql will be needed to drop the data also

isnt too hard to do tbh

**Comment by Pitu** [comment]
This should also apply to album links

**Comment by CyberiumShadow** [comment]
BobbyWibowo's fork has Cloudflare purge and file expiry cleanup implemented but the fork is still on V3 so I'm not sure how well it'd be able to translate to V4

Lack of this functionality is a major blocker for my own image host to migrate to V4

https://github.com/BobbyWibowo/lolisafe/blob/safe.fiery.me/controllers/utilsController.js#L513
https://github.com/BobbyWibowo/lolisafe/blob/safe.fiery.me/controllers/utilsController.js#L622

and some relevant scripts 
https://github.com/BobbyWibowo/lolisafe/tree/safe.fiery.me/scripts

**Comment by Pitu** [comment]
Definitely something to look forward to, that's super useful

**Comment by test137E29B** [comment]
Added File Expiry with https://github.com/WeebDev/chibisafe/pull/307

Design decisions will need to be made though between:
- Potentially leaving files behind  OR
- Much Higher Database operations  OR
- Potential files not existing for the duration of a delete when requested by a client

Have to choose which of these is the lesser evil. For most people I imagine higher db ops is fine, as it's only them or a few people using it, but for a larger host with constant file uploads / short expiry times, it would put a fair amount of load on the db.

IMO option 2 with db ops being higher is the way to go, but it's inefficient. Open to other opinions.

**Comment by kattjevfel** [comment]
In the above linked PR, it was mentioned 

> The new version is way more extensible and this is in the roadmap!
> That being said, we're trying to release a new feature weekly so this could be tackled in the following weeks

this was now over a year ago, was there any progress on this, or should I give up hope on using this software? :P (not meant to be offensive)

**Comment by Pitu** [comment]
There was a lot of progress but this feature didn't make it to the v6 release.

**Comment by PlexSheep** [comment]
What is the status of this? Just hosted it on my server and I mainly want to use it to upload random stuff, send links to others, and have it be removed after 90 days or so.

**Comment by iceleo-com** [comment]
Hi, I think this feature is useful, hope it will be in the next update

I have to program a sub-program to schedule the deletion of images for now
For users who don't have much programming skills, implementing a scheduled deletion program is impossible

**Comment by Pitu** [comment]
This feature is still not in chibisafe, although there are plans on adding it on the next major release but there is no ETA at the moment
